app/waf/src/models/adc.model.ts

```diff
@@ -16,23 +16,30 @@ export class Adc extends Entity {
 
   @property({
     type: 'string',
+    required: true,
   })
-  host?: string;
+  host: string;
 
   @property({
     type: 'number',
+    required: false,
+    default: 443,
   })
-  port?: number;
+  port: number;
 
   @property({
     type: 'string',
+    required: false,
+    default: 'admin',
   })
-  username?: string;
+  username: string;
 
   @property({
     type: 'string',
+    required: false,
+    default: 'admin',
```

Comment:
why do we need default value...
It sounds weird then user succeed without inputting any credential information.
So left it to be required. or let bigip response the unauth* message is better..

Reply:
We need to modify this model, after we implement trusted group. Let it go in this way now.